js/piece.js

```diff
@@ -138,4 +139,33 @@ export class Piece {
     {
         throw new Error("Method 'possibleMoves()' must be implemented");
     }
-}
\ No newline at end of file
+
+    /**
+     * Converts cell id into row and column number
+     * @param id cell id
+     * @returns {(number|number)[]}
+     */
+    convertId(id) {
+        let row = id.charCodeAt(0) - 97;
+        let col = parseInt(id.charAt(1));
+        col = Math.abs(col - 8);
+        return [row, col];
+    }
+
+    /**
+     * Converts piece position into id
+     * @returns {string}
+     */
+    convertPos() {
+        return String.fromCharCode(this.rowPos + 97) + Math.abs(this.colPos - 8);
+    }
+
+    validateMove(row, col, possibleMoves) {
+        for (let i = 0; i < possibleMoves.length; i++) {
+            if (possibleMoves[i][0] === row && possibleMoves[i][1] === col) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
If you want, the code on lines `164 - 169` can be simplified by using the `Array.includes()` method.

Reply:
I originally tried using the Array.includes() method but because it is searching a 2d array it did not behave as desired.